validator_derive/src/lib.rs

```diff
@@ -208,112 +211,153 @@ fn quote_field_validations(
+                            : only a string is allowed"),
+                        };
+                    },
+                    "arg" => {
+                        match lit_to_string(lit) {
+                            Some(s) => {
+                                match syn::parse_str::<syn::Type>(s.as_str()) {
+                                    Ok(arg_type) => {
+                                        assert_custom_arg_type(&lit.span(), &arg_type);
+                                        args = Some(CustomArgument::new(lit.span(), arg_type));
                                     }
-                                    None => error(lit.span(), "invalid argument type for `arg` of `custom` validator: expected a string")
-                                };
-                            },
-                            _ => error(lit.span(), "Unknown argument")
-                        }
-                    } else {
-                        error(arg.span(), "Unexpected args")
-                    }
+                                    Err(_) => {
+                                        let mut msg = "invalid argument type for `arg` of `schema` validator: The string has to be a single type.".to_string();
+                                        msg.push_str("\n(Tip: You can combine multiple types into one tuple.)");
+                                        error(lit.span(), msg.as_str());
+                                    }
+                                }
+                            }
+                            None => error(lit.span(), "invalid argument type for `arg` of `custom` validator: expected a string")
+                        };
+                    },
+                    _ => error(lit.span(), "Unknown argument")
                 }
+            } else {
+                error(arg.span(), "Unexpected args")
             }
+        }
+    }
 
-            if function.is_empty() {
-                error(path.span(), "`function` is required");
-            }
+    if function.is_empty() {
+        error(path.span(), "`function` is required");
+    }
 
-            SchemaValidation {
-                function,
-                args,
-                skip_on_field_errors,
-                code,
-                message,
-            }
-        } else {
-            error(attr.span(), "Unexpected struct validator")
-        }
+    SchemaValidation {
+        function,
+        args,
+        skip_on_field_errors,
+        code,
+        message,
     }
 }
 
 /// Finds all struct schema validations
-fn find_struct_validations(struct_attrs: &[syn::Attribute]) -> Vec<SchemaValidation> {
-    struct_attrs
-        .iter()
-        .filter(|attribute| attribute.path == parse_quote!(validate))
-        .map(find_struct_validation)
-        .collect()
+fn collect_struct_validations(struct_attrs: &[syn::Attribute]) -> (Vec<SchemaValidation>, bool) {
```

Comment:
This is the actual change, most code is just for handling the many cases of meta types (similarly to `collect_field_validations`)